packages/serializer/addon/json-api.js

```diff
@@ -471,6 +471,170 @@ const JSONAPISerializer = JSONSerializer.extend({
     return dasherize(key);
   },
 
+  // SERIALIZE
+  /**
+    Called when a record is saved in order to convert the
+    record into JSON.
+
+    For example, consider this model:
+
+    ```app/models/comment.js
+    import Model, { attr, belongsTo } from '@ember-data/model';
+
+    export default class CommentModel extends Model {
+      @attr title;
+      @attr body;
+
+      @belongsTo('user') author;
+    }
+    ```
+
+    The default serialization would create a JSON-API resource object like:
+
+    ```javascript
+    {
+      "data": {
+        "type": "comments",
+        "attributes": {
+          "title": "Rails is unagi",
+          "body": "Rails? Omakase? O_O",
+        },
+        "relationships": {
+          "author": {
+            "data": {
+              "id": "12",
+              "type": "users"
+            }
+          }
+        }
+      }
+    }
+    ```
+
+    By default, attributes are passed through as-is, unless
+    you specified an attribute type (`attr('date')`). If
+    you specify a transform, the JavaScript value will be
+    serialized when inserted into the JSON hash.
+
+    Belongs-to relationships are converted into JSON-API
+    resource identifier objects.
+
+    ## IDs
+
+    `serialize` takes an options hash with a single option:
+    `includeId`. If this option is `true`, `serialize` will,
+    by default include the ID in the JSON object it builds.
+
+    The adapter passes in `includeId: true` when serializing
```

Comment:
```suggestion
    The JSONAPIAdapter passes in `includeId: true` when serializing a record
    for `createRecord` or `updateRecord`.
```

This is true for both, the `rest` adapter sets it to true for `createRecord` which we inherit, and the json-api adapter sets it to true (by omission of last arg to the private `serializeIntoHash` helper) for `updateRecord`